framework/src/org/apache/cordova/CordovaPlugin.java

```diff
@@ -40,6 +42,7 @@ Licensed to the Apache Software Foundation (ASF) under one
     public CordovaInterface cordova;
     protected CordovaPreferences preferences;
     private String serviceName;
+    protected String [] permissions;
```

Comment:
Each of the core plugins are creating their own version of this field either on the instance or `static`. This can be quite confusing. How would you explain to a plugin developer - what they are expected to do with this field.